test/common/http/http1/codec_impl_test.cc

```diff
@@ -3904,5 +3880,301 @@ TEST_P(Http1ClientConnectionImplTest, InvalidResponseFirstCharacter) {
   EXPECT_EQ(1u, buffer.length());
 }
 
+// A first read of zero bytes when parsing a request is ignored.
+TEST_P(Http1ServerConnectionImplTest, FirstReadEOF) {
+  initialize();
+  InSequence s;
+
+  Buffer::OwnedImpl empty;
+  auto status = codec_->dispatch(empty);
+  ASSERT_TRUE(status.ok());
+
+  StrictMock<MockRequestDecoder> decoder;
```

Comment:
optional: Alright to create the StrictMock decoder above the first dispatch of the empty data? That would drive home the verification that the first empty read is ignored.